app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,63 @@
+class PaymentsController < ApplicationController
+  before_action :set_payment, only: [:edit, :update, :destroy]
+  before_action :validates_role, only: [:new, :edit, :update, :create]
+  before_action :validates_admin, only: :destroy
+
+  def new
+    find_project
+    @payment = Payment.new
+  end
+
+  def create
+    @payment = Payment.new(payment_params)
```

Comment:
create a before action to find project and then initialise payment via that project.
This way you also know if the project_id is valid or not, as incase of it not being found we won't get to this point.

Reply:
method is already there, just utilise it via before action.